crates/sozo/src/ops/migration/mod.rs

```diff
@@ -57,21 +58,73 @@ where
     // Calculate diff between local and remote World manifests.
 
     ui.print_step(2, "🧰", "Evaluating Worlds diff...");
-    let diff = WorldDiff::compute(local_manifest, remote_manifest);
+    let diff = WorldDiff::compute(local_manifest.clone(), remote_manifest.clone());
     let total_diffs = diff.count_diffs();
     ui.print_sub(format!("Total diffs found: {total_diffs}"));
 
     if total_diffs == 0 {
         ui.print("\n✨ No changes to be made. Remote World is already up to date!")
     } else {
         // Mirate according to the diff.
-        apply_diff(ws, target_dir, diff, name, world_address, &account, Some(args.transaction))
+        let world_address = apply_diff(
+            ws,
+            &target_dir,
+            diff,
+            name,
+            world_address,
+            &account,
+            Some(args.transaction),
+        )
+        .await?;
+
+        update_world_manifest(ws, local_manifest, remote_manifest, target_dir, world_address)
             .await?;
     }
 
     Ok(())
 }
 
+async fn update_world_manifest<U>(
+    ws: &Workspace<'_>,
+    mut local_manifest: Manifest,
+    remote_manifest: Option<Manifest>,
+    target_dir: U,
+    world_address: FieldElement,
+) -> Result<()>
+where
+    U: AsRef<Path>,
+{
+    let ui = ws.config().ui();
+    ui.print("\n✨ Updating manifest.json...");
+    local_manifest.world.address = Some(world_address);
+
+    let base_class_hash = match remote_manifest {
+        Some(manifest) => manifest.base.class_hash,
+        None => local_manifest.base.class_hash,
+    };
+
+    local_manifest.contracts.iter_mut().for_each(|c| {
+        let salt = poseidon_hash_many(
+            &c.name
+                .chars()
+                .collect::<Vec<_>>()
+                .chunks(31)
+                .map(|chunk| {
+                    let s: String = chunk.iter().collect();
+                    cairo_short_string_to_felt(&s).unwrap()
+                })
+                .collect::<Vec<_>>(),
```

Comment:
We should move this to a helper function since it is used in a few places now

Reply:
yes, done